apps/omg_watcher/lib/omg_watcher/exit_processor.ex

```diff
@@ -268,64 +269,46 @@ defmodule OMG.Watcher.ExitProcessor do
-        {:stop, :unknown_piggybacks, state}
-
-      {:unknown_in_flight_exit, unknown_ifes} ->
-        _ = Logger.error("Unknown in-flight exits: #{inspect(unknown_ifes)}")
-        {:stop, :unknown_in_flight_exit, Elixir.Agent.Server, state}
-    end
+    {:ok, state, db_updates} = Core.finalize_in_flight_exits(state, finalizations)
+    {:reply, {:ok, db_updates}, state}
   end
 
   @doc """
   Combine data from `ExitProcessor` and `OMG.State` to figure out what to do about exits
   """
   def handle_call(:check_validity, _from, state) do
-    {state1, request} = prepare_validity_check(state)
-    {chain_status, events} = Core.invalid_exits(request, state1)
-    {:reply, {chain_status, events}, state}
+    state1 =
+      %ExitProcessor.Request{}
+      |> update_with_ife_txs_from_blocks(state)
+
+    response =
+      %ExitProcessor.Request{}
+      |> fill_request_with_data(state1)
+      |> Core.check_validity(state1)
+
+    {:reply, response, state1}
   end
 
   def handle_call(:get_active_in_flight_exits, _from, state),
     do: {:reply, {:ok, Core.get_active_in_flight_exits(state)}, state}
 
   def handle_call({:get_competitor_for_ife, txbytes}, _from, state) do
-    # NOTE: future of using `ExitProcessor.Request` struct not certain, see that module for details
+    # TODO: run_status_gets and getting all non-existent UTXO positions imaginable can be optimized out heavily
+    #       only the UTXO positions being inputs to `txbytes` must be looked at, but it becomes problematic as
+    #       txbytes can be invalid so we'd need a with here...
     competitor_result =
       %ExitProcessor.Request{}
-      # TODO: run_status_gets and getting all non-existent UTXO positions imaginable can be optimized out heavily
-      #       only the UTXO positions being inputs to `txbytes` must be looked at, but it becomes problematic as
-      #       txbytes can be invalid so we'd need a with here...
-      |> run_status_gets()
-      |> Core.determine_utxo_existence_to_get(state)
-      |> run_utxo_exists()
-      |> Core.determine_spends_to_get(state)
-      |> run_spend_getting()
-      |> Core.determine_blocks_to_get()
-      |> run_block_getting()
+      |> fill_request_with_data(state)
       |> Core.get_competitor_for_ife(state, txbytes)
 
     {:reply, competitor_result, state}
   end
 
   def handle_call({:prove_canonical_for_ife, txbytes}, _from, state) do
-    # NOTE: future of using `ExitProcessor.Request` struct not certain, see that module for details
+    # TODO: same comment as above in get_competitor_for_ife
     canonicity_result =
       %ExitProcessor.Request{}
-      # TODO: same comment as above in get_competitor_for_ife
-      |> run_status_gets()
-      |> Core.determine_utxo_existence_to_get(state)
-      |> run_utxo_exists()
-      |> Core.determine_spends_to_get(state)
-      |> run_spend_getting()
-      |> Core.determine_blocks_to_get()
-      |> run_block_getting()
+      |> fill_request_with_data(state)
```

Comment:
what about `create_request_environment` or sth similar?

Reply:
see the explicitness/readability argument above